src/Parser/Anime/EpisodeListItemParser.php

```diff
@@ -118,12 +118,18 @@ public function getAired(): ?\DateTimeImmutable
 
 
     /**
-     * @return float
+     * @return float|null
      */
-    public function getScore(): float
+    public function getScore(): ?float
     {
-        return (float) $this->crawler
-            ->filterXPath('//td[contains(@class, \'episode-poll\')]/div[contains(@class, "average")]/span')
+        $node = $this->crawler
+            ->filterXPath('//td[contains(@class, \'episode-poll\')]/div[contains(@class, "average")]/span');
+
+        if (!$node->count()) {
+            return null;
+        }
+
+        return (float) $node
```

Comment:
what if the contents are not castable to float? Is there a way to safe cast it here and throw a meaningful exception if the cast fails?

Reply:
@pushrbx Hmm I'm not sure. PHP enforces float casting on any string. If float fails it returns `0` instead - which I think could be a value somewhere.

I'm using the same logic for anime/manga score properties btw
https://github.com/jikan-me/jikan/blob/88339995b29b69b45b33062ca7cd056665169da7/src/Parser/Anime/AnimeParser.php#L557-L567

Reply:
> If float fails it returns 0 instead
That sounds a bit bad I think. The parser should be vocal about unexpected things, it should not swallow problems imo.

Reply:
By fail I mean, it forces anything it doesn't understand into 0. I think the tests should cover anything unexpected here?

Reply:
Well yes. I'm only suggesting a minor QoL improvement. Imagine something goes wrong then you have to debug it because it was 0, and the assert fails in the test: "4.4 not equals to 0". You will scratch your head at first why it's 0, then you do some magic in the debugger with expression evaluator to figure out what's inside the DOM, then you realize that something is wrong with the value. You might have spent 10-15 minutes with it by that time. If it would just throw an exception saying x value was there instead of y, you don't have to dive into the debugger.

Reply:
Something like this perhaps? https://stackoverflow.com/a/56851835